### PR TITLE
Bluetooth: HCI_UART: Update main stack and refactor conf files

### PR DESCRIPTION
**Pull request by joerchan**
For Nordic alternative HCI driver the main stack usage caused stack overflow issues.
Refactor conf files to recommended CONF_FILE usage as the current CONF_FILE mechanism does not work when providing extra CONF_FILE or CONF_OVERLAY. Configurations will be lost in this case.

### CODE REVIEW

**Comment by joerchan** [comment]
CC: @rugeGerritsen

File: samples/bluetooth/hci_uart/src/main.c [hunk 1]
```diff
@@ -400,5 +402,25 @@ void main(void)
...
+			STACK_ANALYZE("idle thread stack",
+				      z_idle_stack);
+			STACK_ANALYZE("sys workqueue stack",
+				      sys_work_q_stack);
+			prio_ts = k_uptime_get_32();
+		}
+
+		if (k_uptime_get_32() - prio_ts > K_SECONDS(5)) {
+			prio_ts = k_uptime_get_32();
+		}
```
Comment (jhedberg) [inline samples/bluetooth/hci_uart/src/main.c]: This whole branch seems unnecessary to me. The condition is identical to the branch above it and you're not doing any long-duration work in that branch.

Comment (joerchan) [reply samples/bluetooth/hci_uart/src/main.c]: yeah, removed it.

**Comment by cvinayak** [comment]
We should address this at kernel level and not have these in individual samples. For some context, see: https://github.com/zephyrproject-rtos/zephyr/pull/15856

@ceolin FYI.

File: samples/bluetooth/hci_uart/src/main.c [hunk 2]
```diff
@@ -400,5 +402,21 @@ void main(void)
 		if (err) {
 			LOG_ERR("Failed to send");
 		}
+
+#if defined(CONFIG_INIT_STACKS)
+		static u32_t prio_ts;
+
+		if (k_uptime_get_32() - prio_ts > K_SECONDS(5)) {
+			STACK_ANALYZE("main thread stack",
```
Comment (andrewboie) [inline samples/bluetooth/hci_uart/src/main.c]: STACK_ANALYZE is deprecated, please do not use it.
You want a k_thread_foreach() loop and call k_thread_stack_space_get() on each thread.

File: include/debug/stack.h [hunk 3]
```diff
@@ -15,6 +15,12 @@
 #include <logging/log.h>
 #include <stdbool.h>
 
+extern K_THREAD_STACK_DEFINE(z_main_stack, CONFIG_MAIN_STACK_SIZE);
```
Comment (andrewboie) [inline include/debug/stack.h]: do not put these here, they are private to the kernel

**Comment by andrewboie** [comment]
you should really separate out the patches which are strictly bug fixes into a separate PR to get them in faster.

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
needs changes

**Comment by joerchan** [comment]
@andrewboie Your suggestion does not give me the stack usage of the threads mentioned, with the k_thread_foreach(_unlocked) I only get "BT RX" and "BT RX pri" threads. Not the main stack or any of the others.

Another thing I noticed is that k_thread_foreach does not iterate over the static threads either, but adding code for that did not give me main stack usage either.

The application sets the stack size of these threads and performs a lot of the code that runs it them. But don't have a way to analyze the stack usage for these threads. 

What about adding `struct k_thread_t *k_thread_get(const char *name)` functions that would allow us to analyze these threads?

File: samples/bluetooth/hci_uart/src/main.c (hunk 2 above)
Comment (joerchan) [reply samples/bluetooth/hci_uart/src/main.c]: Will replace STACK_ANALYZE usage. But need a way to do it for main stack. k_thread_foreach will not give me main stack.

File: include/debug/stack.h (hunk 3 above)
Comment (joerchan) [reply include/debug/stack.h]: I need an alternative so that I can debug the main stack.

**Comment by joerchan** [comment]
@andrewboie Sorry for the noise, it appears to be a logging issue. I only get some of the LOG_INF statements. Using printk I see it all.

File: include/debug/stack.h (hunk 3 above)
Comment (joerchan) [reply include/debug/stack.h]: Removed

File: samples/bluetooth/hci_uart/src/main.c (hunk 2 above)
Comment (joerchan) [reply samples/bluetooth/hci_uart/src/main.c]: Replaced with suggestion.

File: samples/bluetooth/hci_uart/debug.conf [hunk 4]
```diff
@@ -0,0 +1,22 @@
+CONFIG_ASSERT=y
```
Comment (carlescufi) [inline samples/bluetooth/hci_uart/debug.conf]: Could you please update the sample's `README.rst` with info on how to use this `debug.conf`?

**Comment by joerchan** [comment]
Split the stack debug changes into a new PR.
This is now only the main stack usage update and refactor of the conf files.
The refactor of the conf files is needed to easily add additional configuration options from debug conf file.

**Review by carlescufi — APPROVED** [review]
LGTM, thanks!

**Comment by joerchan** [comment]
@andrewboie The things you objected to have been removed from the PR.

**Review by jhedberg — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request merged**